src/main/java/org/opensearch/neuralsearch/processor/TextImageEmbeddingProcessor.java

```diff
@@ -0,0 +1,218 @@
+        Map<String, String> mapWithKnnKeys = new LinkedHashMap<>();
+        for (Map.Entry<String, String> fieldMapEntry : fieldMap.entrySet()) {
+            String originalKey = fieldMapEntry.getValue(); // field from ingest document that we need to sent as model input, part of
+                                                           // processor definition
+
+            if (!sourceAndMetadataMap.containsKey(originalKey)) {
+                continue;
+            }
+            if (!(sourceAndMetadataMap.get(originalKey) instanceof String)) {
+                throw new IllegalArgumentException("Unsupported format of the field in the document, value must be a string");
+            }
+            mapWithKnnKeys.put(originalKey, (String) sourceAndMetadataMap.get(originalKey));
+        }
+        return mapWithKnnKeys;
+    }
+
+    @SuppressWarnings({ "unchecked" })
+    @VisibleForTesting
+    Map<String, Object> buildTextEmbeddingResult(String knnKey, List<Float> modelTensorList) {
+        Map<String, Object> result = new LinkedHashMap<>();
+        result.put(knnKey, modelTensorList);
+        return result;
+    }
+
+    private void validateEmbeddingFieldsValue(IngestDocument ingestDocument) {
+        Map<String, Object> sourceAndMetadataMap = ingestDocument.getSourceAndMetadata();
+        for (Map.Entry<String, String> embeddingFieldsEntry : fieldMap.entrySet()) {
+            Object sourceValue = sourceAndMetadataMap.get(embeddingFieldsEntry.getKey());
+            if (sourceValue != null) {
+                String sourceKey = embeddingFieldsEntry.getKey();
+                Class<?> sourceValueClass = sourceValue.getClass();
+                if (List.class.isAssignableFrom(sourceValueClass) || Map.class.isAssignableFrom(sourceValueClass)) {
+                    validateNestedTypeValue(sourceKey, sourceValue, () -> 1);
+                } else if (!String.class.isAssignableFrom(sourceValueClass)) {
+                    throw new IllegalArgumentException("field [" + sourceKey + "] is neither string nor nested type, can not process it");
+                } else if (StringUtils.isBlank(sourceValue.toString())) {
+                    throw new IllegalArgumentException("field [" + sourceKey + "] has empty string value, can not process it");
+                }
+            }
+        }
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private void validateNestedTypeValue(String sourceKey, Object sourceValue, Supplier<Integer> maxDepthSupplier) {
+        int maxDepth = maxDepthSupplier.get();
+        if (maxDepth > MapperService.INDEX_MAPPING_DEPTH_LIMIT_SETTING.get(environment.settings())) {
+            throw new IllegalArgumentException("map type field [" + sourceKey + "] reached max depth limit, can not process it");
+        } else if ((List.class.isAssignableFrom(sourceValue.getClass()))) {
+            validateListTypeValue(sourceKey, sourceValue);
```

Comment:
With update in `validateListTypeValue` mentioned later
```suggestion
            validateListTypeValue(sourceKey, (List)sourceValue);
```

Reply:
ack